optimum/onnxruntime/modeling_decoder.py

```diff
@@ -465,121 +484,214 @@ def _from_pretrained(
         config: "PretrainedConfig",
         use_auth_token: Optional[Union[bool, str]] = None,
         revision: Optional[str] = None,
-        force_download: bool = True,
+        force_download: bool = False,
         cache_dir: Optional[str] = None,
         decoder_file_name: str = ONNX_DECODER_NAME,
         decoder_with_past_file_name: str = ONNX_DECODER_WITH_PAST_NAME,
         subfolder: str = "",
         local_files_only: bool = False,
         use_cache: bool = True,
-        use_io_binding: bool = True,
         provider: str = "CPUExecutionProvider",
         session_options: Optional[onnxruntime.SessionOptions] = None,
         provider_options: Optional[Dict[str, Any]] = None,
+        use_io_binding: bool = True,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
     ):
-        file_names = {}
-        # Load model from a local directory
-        if os.path.isdir(os.path.join(model_id, subfolder)):
-            decoder_with_past_path = (
-                os.path.join(model_id, subfolder, decoder_with_past_file_name) if use_cache else None
+        model_path = Path(model_id)
+
+        def validate_filename(filename):
```

Comment:
Would this be better out of `_from_pretrained`. Otherwise this is redefined every time `_from_pretrained` is called, am I correct?

Reply:
why have it as a function in the first time then?

Reply:
We use it multiple time inside this function, but never outside. We could make that a method, but that is not needed specifically. While it is true that there is redundancy between this and `modeling_seq2seq.py`, I wanted to keep the function simple (making it a method woulrd require more arguments and so on).